alacritty_terminal/src/term/mod.rs

```diff
@@ -289,6 +290,14 @@ impl<'a, C> RenderableCellsIter<'a, C> {
             None => return false,
         };
 
+        // Do not invert block cursor at selection boundaries
+        if self.cursor.style == CursorStyle::Block
+            && self.cursor.point == point
+            && (selection.start == point || selection.end == point)
+        {
```

Comment:
I didn't actually consider the four quadrants of block selection.

The idea here is that the block cursor doesn't make the selection non-deterministic. If you always invert the block cursor, it's impossible to tell if the cursor is behind the selection, or right at the end of it. As a result, it's impossible to tell what will be copied just by looking at a picture of a selection with a block cursor.

This is especially bad when your selection is moved around with a block cursor, since this will mean that the cursor itself is always 'invisible'.